### PR TITLE
Add retention duraction, retain acked message to pubsub

### PR DESCRIPTION
**Pull request by rileykarson**
Fixes https://github.com/terraform-providers/terraform-provider-google/issues/2264

/cc @rambleraptor @slevenick this will make downstreams for y'all I think

-----------------------------------------------------------------
# [all]

### CODE REVIEW

File: products/pubsub/api.yaml [hunk 1]
```diff
@@ -144,3 +144,23 @@ objects:
...
+          of acknowledged messages, and thus configures how far back in time a
+          subscriptions.seek can be done. Defaults to 7 days. Cannot be more
+          than 7 days (`"604800s"`) or less than 10 minutes (`"600s"`).
+
+          A duration in seconds with up to nine fractional digits, terminated
+          by 's'. Example: `"600.5s"`.
+      - !ruby/object:Api::Type::Boolean
+        name: 'retainAckedMessages'
+        description: |
+          Indicates whether to retain acknowledged messages. If `true`, then
```
Comment (chrisst) [inline products/pubsub/api.yaml]: I assume this defaults to false, but I think it would be valuable to be explicit for users.

Comment (rileykarson) [reply products/pubsub/api.yaml]: I'm 👎, false defaults should remain unspecified and the description implies that it needs to be set to true imo.

Comment (chrisst) [reply products/pubsub/api.yaml]: I look at it from the perspective of describing that there is behavior of retaining or purging ack'ed messages, in which case the default behavior is only clear for people used to streaming queues. And IMO there's no harm in being explicit over implicit

Comment (chrisst) [reply products/pubsub/api.yaml]: For those who are familiar with Go/Terraform boolean behavior or with streaming providers you are right it's implied.

**Pull request merged**